File: frontend/src/application/dto/job.dto.ts
import { Job, JobType, LocationType, JobStatus } from '@/domain/entities/job.entity'

export interface CreateJobRequest {
  startup_id: string
  title: string
  description: string
  requirements: string
  job_type: JobType
  location_type: LocationType
  city?: string
  country: string
  salary_min?: number
  salary_max?: number
  currency: string
  application_url?: string
  application_email?: string
  expires_at?: string
}

export interface UpdateJobRequest {
  id: string
  title?: string
  description?: string
  requirements?: string
  job_type?: JobType
  location_type?: LocationType
  city?: string
  country?: string
  salary_min?: number
  salary_max?: number
  currency?: string
  application_url?: string
  application_email?: string
  status?: JobStatus
  expires_at?: string
}

export interface JobListFilters {
  startup_id?: string
  job_type?: JobType
  location_type?: LocationType
  status?: JobStatus
  search?: string
  country?: string
  city?: string
  salary_min?: number
  salary_max?: number
  currency?: string
  order_by?: string
  order_dir?: 'ASC' | 'DESC'
  page?: number
  page_size?: number
}

export type JobResponse = Job

File: backend/internal/application/dto/job_dto.go
package dto

type CreateJobInput struct {
	StartupID        string  `json:"startup_id" validate:"required"`
	Title            string  `json:"title" validate:"required,min=5,max=100"`
	Description      string  `json:"description" validate:"required,min=20"`
	Requirements     string  `json:"requirements" validate:"required,min=10"`
	JobType          string  `json:"job_type" validate:"required,oneof=full_time part_time contract internship"`
	LocationType     string  `json:"location_type" validate:"required,oneof=remote hybrid onsite"`
	City             string  `json:"city"`
	Country          string  `json:"country" validate:"required"`
	SalaryMin        *int    `json:"salary_min" validate:"omitempty,min=0"`
	SalaryMax        *int    `json:"salary_max" validate:"omitempty,min=0"`
	Currency         string  `json:"currency" validate:"required,len=3"`
	ApplicationURL   *string `json:"application_url" validate:"omitempty,url"`
	ApplicationEmail *string `json:"application_email" validate:"omitempty,email"`
	ExpiresAt        *string `json:"expires_at" validate:"omitempty"`
}

type UpdateJobInput struct {
	ID               string  `json:"id"`
	Title            *string `json:"title" validate:"omitempty,min=5,max=100"`
	Description      *string `json:"description" validate:"omitempty,min=20"`
	Requirements     *string `json:"requirements" validate:"omitempty,min=10"`
	JobType          *string `json:"job_type" validate:"omitempty,oneof=full_time part_time contract internship"`
	LocationType     *string `json:"location_type" validate:"omitempty,oneof=remote hybrid onsite"`
	City             *string `json:"city"`
	Country          *string `json:"country"`
	SalaryMin        *int    `json:"salary_min" validate:"omitempty,min=0"`
	SalaryMax        *int    `json:"salary_max" validate:"omitempty,min=0"`
	Currency         *string `json:"currency" validate:"omitempty,len=3"`
	ApplicationURL   *string `json:"application_url" validate:"omitempty,url"`
	ApplicationEmail *string `json:"application_email" validate:"omitempty,email"`
	Status           *string `json:"status" validate:"omitempty,oneof=active closed"`
	ExpiresAt        *string `json:"expires_at" validate:"omitempty"`
}

type JobOutput struct {
	ID               string  `json:"id"`
	StartupID        string  `json:"startup_id"`
	StartupName      string  `json:"startup_name,omitempty"`
	StartupSlug      string  `json:"startup_slug,omitempty"`
	Title            string  `json:"title"`
	Description      string  `json:"description"`
	Requirements     string  `json:"requirements"`
	JobType          string  `json:"job_type"`
	LocationType     string  `json:"location_type"`
	City             string  `json:"city"`
	Country          string  `json:"country"`
	SalaryMin        *int    `json:"salary_min"`
	SalaryMax        *int    `json:"salary_max"`
	Currency         string  `json:"currency"`
	ApplicationURL   *string `json:"application_url"`
	ApplicationEmail *string `json:"application_email"`
	Status           string  `json:"status"`
	ExpiresAt        *string `json:"expires_at"`
	CreatedAt        string  `json:"created_at"`
	UpdatedAt        string  `json:"updated_at"`
}
